Godeps/_workspace/src/k8s.io/kubernetes/pkg/kubectl/cmd/util/factory.go

```diff
@@ -597,21 +613,22 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 	}
 }
 
-// GetFirstPod returns the first pod of an object from its namespace and selector and the number of matching pods
+// GetFirstPod returns a pod matching the namespace and label selector
+// and the number of all pods that match the label selector.
 func GetFirstPod(client *client.Client, namespace string, selector labels.Selector) (*api.Pod, int, error) {
+	options := api.ListOptions{LabelSelector: selector}
 	var pods *api.PodList
-	for pods == nil || len(pods.Items) == 0 {
+	// TODO: Make timeout configurable
+	if err := wait.PollImmediate(2*time.Second, 20*time.Second, func() (bool, error) {
 		var err error
-		options := api.ListOptions{LabelSelector: selector}
 		if pods, err = client.Pods(namespace).List(options); err != nil {
 			return nil, 0, err
 		}
-		if len(pods.Items) == 0 {
-			time.Sleep(2 * time.Second)
-		}
+		return len(pods.Items) > 0, nil
+	}); err != nil {
+		return nil, 0, err
 	}
-	pod := &pods.Items[0]
-	return pod, len(pods.Items), nil
+	return &pods.Items[0], len(pods.Items), nil
```

Comment:
> I was filtering active pods and then sorting to the oldest but then remembered that @deads2k was against it when he was adding this upstream.

I want different sorts for different uses:
 1. `rsh`: pod better be running, but it probably doesn't matter which one.
 2. `logs`: show me the crashlooping one, because all happy families are the same.
 3. `attach`: I probably want the most recent pod.

I haven't looked at this pull, but does it allow different comparators?